src/inventorymanager.cpp

```diff
@@ -473,18 +496,38 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 			to_inv.type == InventoryLocation::NODEMETA &&
 			from_inv.p == to_inv.p) {
 		PLAYER_TO_SA(player)->nodemeta_inventory_OnMove(
-				from_inv.p, from_list, from_i,
-				to_list, to_i, count, player);
+			*this, count, player);
 	} else {
 		// Destination is nodemeta
 		if (to_inv.type == InventoryLocation::NODEMETA) {
 			PLAYER_TO_SA(player)->nodemeta_inventory_OnPut(
-					to_inv.p, to_list, to_i, src_item, player);
+				*this, src_item, player);
 		}
 		// Source is nodemeta
-		else if (from_inv.type == InventoryLocation::NODEMETA) {
+		if (from_inv.type == InventoryLocation::NODEMETA) {
```

Comment:
Little but effective bugfix. The `else` keyword prevented from `on_take` being run when only the destination is a nodemeta inventory.